cmd/juju/action/cancel.go

```diff
@@ -87,14 +87,15 @@ func (c *cancelCommand) Run(ctx *cmd.Context) error {
 		return errors.Errorf("identifier(s) %q matched action(s) %q, but no actions were canceled", c.requestedIds, actionTags)
 	}
 
-	var unCanceledActions []string
+	var unCanceledActions []names.ActionTag
 	var canceledActions []params.ActionResult
-	for _, result := range actions.Results {
-		if result.Error != nil {
-			unCanceledActions = append(unCanceledActions, result.Action.Tag)
-			continue
+
+	for i, result := range actions.Results {
+		if result.Action != nil {
```

Comment:
I agree, having more insight into why the transaction was canceled would be empowering, would not actually add very much complexity to the code, and the API changes necessary would be backward compatible. From what I can tell, the cost of implementing this suggestion is low and especially so compared to the benefit it brings. If you don't mind I will add it to this PR. Thank you @jameinel.